trie/sync_test.go

```diff
@@ -704,7 +704,7 @@ func syncWithHookWriter(t *testing.T, root common.Hash, db ethdb.Database, srcDb
 			syncPath: NewSyncPath([]byte(paths[i])),
 		})
 	}
-	reader, err := srcDb.Reader(root)
```

Comment:
If we want to make the change smaller, how about first a pure refactoring/renaming-PR, changing e.g. this `Reader->NodeReader`, and `Database` to `NodeDatabase`, and other similar pure changes?

Reply:
sounds good to me!